dispatcher/providers/azurebatch.go

```diff
@@ -261,7 +279,13 @@ func (b *AzureBatch) Reconcile() error {
 		}
 
 		if t.State == batch.TaskStateCompleted && *t.ExecutionInfo.ExitCode != 0 {
```

Comment:
nit: this could be folded into the above if - i.e. `if t.State == batch.TaskStateCompleted { if *t.ExecutionInfo.ExitCode == 0 {} else { //assume all other exit codes are error}` - this saves 1 check against the t.State and an additional check against the ExitCode

Reply:
Good call, would read better too